src/vaadin-dialog.html

```diff
@@ -289,6 +391,172 @@
+                case 'n': {
+                  const height = originalBounds.height - (e.pageY - originalMouse.top);
+                  const top = originalBounds.top + (e.pageY - originalMouse.top);
+                  if (height > minimumSize) {
+                    this._setBounds({top, height});
+                  }
+                  break;
+                }
+                case 'e': {
+                  const width = originalBounds.width + (e.pageX - originalMouse.left);
+                  if (width > minimumSize) {
+                    this._setBounds({width});
+                  }
+                  break;
+                }
+                case 's': {
+                  const height = originalBounds.height + (e.pageY - originalMouse.top);
+                  if (height > minimumSize) {
+                    this._setBounds({height});
+                  }
+                  break;
+                }
+                case 'w': {
+                  const width = originalBounds.width - (e.pageX - originalMouse.left);
+                  const left = originalBounds.left + (e.pageX - originalMouse.left);
+                  if (width > minimumSize) {
+                    this._setBounds({left, width});
+                  }
+                  break;
+                }
+              }
+            });
+          };
+          const stopResize = edge => window.removeEventListener('mousemove', resizeEvents[edge]);
+          const resizeEvents = {};
+          const stopResizeEvents = {};
+
+          // Note: edge controls added before corners
+          ['n', 'e', 's', 'w', 'nw', 'ne', 'se', 'sw'].forEach(edge => {
+            const resizer = document.createElement('div');
+            if (edge.length === 1) {
+              resizer.classList.add('edge');
+            }
+            resizer.classList.add('resizer');
+            resizer.classList.add(edge);
+            resizer.addEventListener('mousedown', e => {
+              e.preventDefault();
+              e.stopPropagation();
+              this._moveOverlayToTop();
+              const overlayBounds = overlay.getBoundingClientRect();
+              const containerBounds = this.$.overlay.getBoundingClientRect();
+              originalBounds.top = overlayBounds.top - containerBounds.top;
+              originalBounds.left = overlayBounds.left - containerBounds.left;
+              originalBounds.width = overlayBounds.width;
+              originalBounds.height = overlayBounds.height;
+              originalMouse.top = e.pageY;
+              originalMouse.left = e.pageX;
+              resizeEvents[edge] = e => resize(e, edge);
+              stopResizeEvents[edge] = () => stopResize(edge);
+              window.addEventListener('mousemove', resizeEvents[edge]);
+              window.addEventListener('mouseup', stopResizeEvents[edge]);
+            });
+            overlay.appendChild(resizer);
+          });
+        }
+
+        _setBounds({top, right, bottom, left, width, height}) {
```

Comment:
iirc the original plan was to only apply top, right, bottom and left. Width and height should be redundant in that case?

Reply:
Then your code wouldn't need to adjust the height of the overlay when you resize with the "n" resizer etc

Reply:
We agreed to go with top, left, width and height. Using bottom and right (instead of width/height) would get complicated because you'd have to adjust these values on window resize.